### ISSUE TITLE
Curl terminated due to a fatal signal!

### ISSUE THREAD

**Issue by fishhead108** [quickemu-project/quickemu#957]
#### Expected behaviour

To be able to download Windows (8,10,11)

#### Actual behaviour

```bash
❯ quickget windows 10
Downloading Windows 10...
 - Product edition ID: 2618
Curl terminated due to a fatal signal!
Downloading VirtIO drivers...
windows-10/virtio-win.iso                                         1%[=>                                                                                                                                                    ]   9.06M  7.90MB/s               ^C


❯ quickget windows 11
Downloading Windows 11...
 - Product edition ID: 2935
Curl terminated due to a fatal signal!
Downloading VirtIO drivers...
windows-11/virtio-win.iso                                         6%[++++++++++                                                                                                                                            ]  41.78M   211KB/s               ^C
```

I tried other OS's and they work fine.
```bash
❯ quickget android 9.0
android-9.0-x86/android-x86-9.0-r2.iso                            9%[=============>                                                                                                                                        ]  69.09M  24.3MB/s               ^C

❯ quickget archlinux latest
archlinux-latest/archlinux-2024.03.01-x86_64.iso                  0%[>                                                                                                                                                     ]   7.05M   691KB/s    eta 29m 46s^C

❯ quickget ubuntu-mate 22.04
ubuntu-mate-22.04/ubuntu-mate-22.04.4-desktop-amd64.iso           4%[+++==>                                                                                                                                                ] 143.50M  14.5MB/s    eta 4m 15s ^C

❯ quickget macos catalina
macos-catalina/OpenCore.qcow2                                   100%[=====================================================================================================================================================>]  20.50M  40.6MB/s    in 0.5s
macos-catalina/OVMF_CODE.fd                                     100%[=====================================================================================================================================================>]   1.88M  8.82MB/s    in 0.2s
```

#### Steps to reproduce the behaviour

Run the command `quickget windows 11` on NixOS

#### Quickemu output

```bash
❯ quickget
ERROR! You must specify an operating system.
 - Operating Systems: alma alpine android antix archlinux archcraft arcolinux batocera blendos bodhi bunsenlabs cachyos centos-stream debian deepin devuan dragonflybsd easyos edubuntu elementary endeavouros endless fedora freebsd freedos garuda gentoo ghostbsd haiku holoiso kali kdeneon kolibrios kubuntu linuxlite linuxmint lmde mageia manjaro mxlinux netboot netbsd nixos lubuntu macos openbsd openindiana opensuse oraclelinux peppermint popos porteus reactos rebornos rockylinux siduction slackware solus spiral tails tinycore trisquel truenas-core truenas-scale ubuntu ubuntu-budgie ubuntucinnamon ubuntukylin ubuntu-mate ubuntu-server ubuntustudio ubuntu-unity vanillaos void vxlinux windows xerolinux xubuntu zorin

        You can also use this arguments:
        Only show ISO download URL
            --show-iso-url / -s {distro} {release} [edition]
        Test if ISO is available
            --test-iso-url / -t {distro} {release} [edition]
        Open distro homepage
            --open-distro-homepage / -o {distro}
```

#### Linux Distribution & Kernel

```bash
[nix-shell:~]$ lsb_release -a
No LSB modules are available.
Distributor ID: NixOS
Description:    NixOS 24.05 (Uakari)
Release:        24.05
Codename:       uakari

[nix-shell:~]$ uname -a
Linux lenovo 6.7.7 #1-NixOS SMP PREEMPT_DYNAMIC Fri Mar  1 12:42:00 UTC 2024 x86_64 GNU/Linux
```

**Comment by lj3954** [comment]
I can confirm that Windows downloads have become somewhat unreliable. See: https://github.com/pbatard/Fido/issues/88, https://github.com/pbatard/Fido/commit/4e6f25f35112c82ee554bb8c602ce85bdccdfeb7.

**Comment by badr-elmazaz** [comment]
Hi same here, is there some workaround or a solution to this problem?

**Comment by lj3954** [comment]
> Hi same here, is there some workaround or a solution to this problem?

If the Windows download fails, Quickget will automatically continue downloading the drivers, creating the unattended setup, and creating a config pointing to an ISO called "windows-{RELEASE}.iso" (with {RELEASE} being the version of Windows). You can download the ISO from Microsoft's website, rename it as mentioned, and move it into the VM's directory.

I'm not sure anyone can provide you with a timeframe on when this issue may be fixed.

**Comment by badr-elmazaz** [comment]
> > Hi same here, is there some workaround or a solution to this problem?
> 
> If the Windows download fails, Quickget will automatically continue downloading the drivers, creating the unattended setup, and creating a config pointing to an ISO called "windows-{RELEASE}.iso" (with {RELEASE} being the version of Windows). You can download the ISO from Microsoft's website, rename it as mentioned, and move it into the VM's directory.
> 
> I'm not sure anyone can provide you with a timeframe on when this issue may be fixed.

Thank you very much it worked correctly. Just downloaded the iso manually from `https://www.microsoft.com/en-US/software-download/windows10ISO` and moved it into `windows-10` directory then started them command `quickget windows 10` on the same directory.

**Comment by flexiondotorg** [comment]
Fixed via #1042
